test/src/main/java/org/apache/accumulo/test/functional/ErrorThrowingIterator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.test.functional;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.WrappingIterator;
+
+/**
+ * Iterator used in tests *and* the test class must spawn a new MAC instance for each test since the
+ * timesThrown variable is static.
+ */
+public class ErrorThrowingIterator extends WrappingIterator {
+
+  public static final String TIMES = "error.throwing.iterator.times";
+
+  private static final String MESSAGE = "Exception thrown from ErrorThrowingIterator";
+  private static final RuntimeException ERROR = new RuntimeException(MESSAGE);
+  private static final AtomicInteger timesThrown = new AtomicInteger(0);
+
+  private static int threshold = 0;
+
+  @Override
+  public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options,
+      IteratorEnvironment env) throws IOException {
+    super.init(source, options, env);
+    threshold = Integer.parseInt(options.get(TIMES));
```

Comment:
Wondering if we can add this w/o causing problems for test that working as expected.

```suggestion
    threshold = Integer.parseInt(options.get(TIMES));
    Preconditions.checkState(threshold <= timesThrown.get());
```

Reply:
This won't work as on the initial use of this iterator, timesThrown is zero.

Reply:
oh right, what about the following

```java
Preconditions.checkState(timesThrown.get() <= threshold);
```

Reply:
see f45a9f5